From 790e96e582025f9fa005b1569c91c94e4e8a6962 Mon Sep 17 00:00:00 2001
From: Matt Jolly <Matt.Jolly@footclan.ninja>
Date: Mon, 16 Oct 2023 17:15:06 +1000
Subject: [PATCH] Consume variables that are set

For versions of at-spi2-core >= 2.50.0 the previous logic would
fail to use the `micro` variable, resulting in a gn build error
as it went out of scope.

Now we set and immediately ensure that variables are consumed.
--- a/build/config/linux/atspi2/BUILD.gn
+++ b/build/config/linux/atspi2/BUILD.gn
@@ -17,19 +17,21 @@ if (use_atk) {
                                       "--version-as-components",
                                     ],
                                 "value")
-    major = atspi_version[0]
-    minor = atspi_version[1]
-    micro = atspi_version[2]
 
     # ATSPI 2.49.90 now defines these for us and it's an error for us to
     # redefine them on the compiler command line.
     # See ATSPI 927344a34cd5bf81fc64da4968241735ecb4f03b
-    if (minor < 49 || (minor == 49 && micro < 90)) {
-      defines = [
-        "ATSPI_MAJOR_VERSION=$major",
-        "ATSPI_MINOR_VERSION=$minor",
-        "ATSPI_MICRO_VERSION=$micro",
-      ]
+    minor = atspi_version[1]
+    if (minor < 49) {
+      micro = atspi_version[2]
+      if (micro < 90) {
+        major = atspi_version[0]
+        defines = [
+          "ATSPI_MAJOR_VERSION=$major",
+          "ATSPI_MINOR_VERSION=$minor",
+          "ATSPI_MICRO_VERSION=$micro",
+        ]
+      }
     }
   }
 }
-- 
2.42.0

